[WD-6796] modify sections in /careers/engineering

akbarkz opened the pull request:
## Done

- create 'Apply based on' section for "fast-track" positions in `/careers/engineering`
- modify 'Featured roles' section in `/careers/engineering` according to [Figma designs](https://www.figma.com/file/c4QYGBPZl5hxr5XrziIVZA/23.10-C.com---%2Fcareers%2Fengineering-(Fast-tracks)?type=design&node-id=98-5748&mode=design&t=LRqU0MZl4Xk191vG-4)
- edit the first section in `/careers/engineering` also according to design

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8002/
- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- Open /careers/engineering and check the changes in first three sections
- Open other /careers pages and check that nothing changed

## Issue / Card

Closes https://warthogs.atlassian.net/browse/WD-6796

webteam-app commented:
Demo starting at https://canonical-com-1094.demos.haus

juanruitina commented:
- Please change the link for DevRel to https://canonical.com/careers/5143011 (my bad!)
- Spacing under the hero seems a bit excessive, maybe just remove `.p-section` from the row within?
- Descriptions should be .875rem, wrapping them inside `<small>` as we do in the [meganav](https://ubuntu-com-12939.demos.haus/) seems to do the trick.
- Please add an `<hr>` above "Explore all engineering roles" for small and medium viewports (hidden on desktop) so we differentiate the link from the list of featured roles.

akbarkz commented:
@juanruitina I addressed your comments, ptal.
Regarding this one `Spacing under the hero seems a bit excessive, maybe just remove .p-section from the row within?`:
I decreased the spacing, but I cannot remove the `p-section` because this section is used in all other Departments career pages too. I hope this change is enough.

juanruitina commented:
One last thing in line with our discussion on Mattermost: please tweak list of featured roles so it shows 9 items max as it's now live. (We will look into this next, [ticket on Jira](https://warthogs.atlassian.net/browse/WD-7025). 29 featured roles is a lot, trimming alphabetically is not ideal either.)

Other than that, this looks good to me. Feel free to change to UX+1 once you tweak it.

carkod commented on templates/shared/_engineering-fast-track.html:
```diff
@@ -1,131 +1,123 @@
+				{% if loop.index < 10 %} <p>
+				<a class="p-featured__url p-card--content" href="/careers/{{job.id}}"
+					onclick="dataLayer.push({'event' : 'GAEvent', 'eventCategory' : 'Featured job', 'eventAction' : 'Click', 'eventLabel' : '{{ job.title }}', 'eventValue' : undefined });">
+					{{job.title }}
```
maybe some space before ` job.title`

carkod reviewed (approved):
Thanks. LGTM
